src/movies.js

```diff
@@ -1,22 +1,70 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const allDirectors = [];
+    moviesArray.map((movie) => allDirectors.push(movie.director));
+    return allDirectors
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    if (moviesArray.length === 0) {
+        return 0;
+    }
+    const dramaMovie = moviesArray.filter((movie) => movie.genre.includes("Drama"));
+    const StevenSpielbergMovies = dramaMovie.filter((movie) => movie.director.includes("Steven Spielberg"));
```

Comment:
```suggestion
    const StevenSpielbergMovies = moviesArray.filter((movie) => movie.genre.includes("Drama") && movie.director.includes("Steven Spielberg"));
```

Si te fijas dentro de la condicion del filter puedes meter tantas condiciones que se quieran evaluar!, es decir puedes poner las dos condiciones directamente!, estaba bien lo que has hecho eeee pero asi queda mas directo y solo es un bucle! :D